### PR TITLE
Update to get_item_info() specifically for decks.

### PR DESCRIPTION
**Pull request by Jmeigs1**
Update to get_item_info() specifically for decks..  Bugfix for issue #11435.

### CODE REVIEW

File: crawl-ref/source/externs.h [hunk 1]
```diff
@@ -548,7 +548,7 @@ struct item_def
     {
         // These must all be the same size!
         short plus2;        ///< legacy/generic name for this union
-        short used_count;   ///< the # of times a deck was used
+        short used_count;   ///< the # of times a deck was used.  Note: Is negitive for stacked deck for # remaining
```
Comment (NormalPerson7) [inline crawl-ref/source/externs.h]: `negative` not `negitive`

File: crawl-ref/source/items.cc [hunk 2]
```diff
@@ -4824,36 +4824,12 @@ item_info get_item_info(const item_def& item)
...
-
-            if (info_cards.empty())
-            {
-                // An empty deck would display as BUGGY, so fake a card.
-                info_cards.push_back((char) 0);
-                info_card_flags.push_back((char) 0);
-            }
-            ii.props[CARD_KEY] = info_cards;
-            ii.props[CARD_FLAG_KEY] = info_card_flags;
+            //Fixes an inconsisancy with display of decks
```
Comment (NormalPerson7) [inline crawl-ref/source/items.cc]: `inconsistency` not `inconsisancy`

File: crawl-ref/source/items.cc [hunk 3]
```diff
@@ -4824,36 +4824,12 @@ item_info get_item_info(const item_def& item)
...
+            //Cards are flagged as seen and all the corresponding deck classes take this into account
```
Comment (NormalPerson7) [inline crawl-ref/source/items.cc]: This line (and the one above) is too long -- they should be maximum 80 characters wide, so split this line after classes please.

**Comment by aidanholm** [comment]
Merged manually at f65d00e5d6; thanks for the PR and thanks @NormalPerson7 for reviewing it

**Pull request closed without merging**